### ISSUE TITLE
Ensure BSK_RL is compatible with BSK installation

### ISSUE THREAD

**Issue by Mark2000** [AVSLab/bsk_rl#89]
Relevant to #14, since our released versions should probably match certain bsk releases

I think we should check on import that a minimum version of bsk_rl is installed (by version? by commit hash for develop?) and warn the user if it is not

**Comment by jbbennett91** [comment]
Any chance this is related to the value mismatch that occurred sometime last week?

Installing current versions of `basilisk` and `bsk_rl`, then testing with `bsk_rl/examples/general_satellite_tasking/single_sat.py` returns the following issue:

<img width="548" alt="image" src="https://github.com/AVSLab/bsk_rl/assets/145808306/0f6f3709-6c9d-4c28-bdbf-c35a717cd350">

**Comment by Mark2000** [comment]
Ah, good catch. I believe that's a result of #90, which adds a flag to the info sometimes that's not in the form `(time, message)`. I'll open a PR tomorrow to address this.

A documentation system is currently being put together and that'll change how example scripts are formatted and handled. Currently, example scripts aren't being caught in the test suite; I'll make sure that they are tested when we make that change so that PRs that break common use patterns like this are avoided in the future.

**Comment by Mark2000** [comment]
@LorenzzoQM, could you review #92 which refactors the previous PR?

**Comment by Mark2000** [comment]
@jbbennett91 Should be resolved now with #92 . As an additional heads up, there's a PR that'll be switching us to a src-style package coming up, so you may need to uninstall and reinstall the package.

**Comment by Mark2000** [comment]
Over in Basilisk, @patkenneally is looking into automated beta version number increasing (so that each commit increases bsk version to 2.2.1b1, 2.2.1b2, 2.2.1b3, etc). This will allow us to match our develop to specific bsk versions without needing to dig around for commit hashes.

Considering code like this in the `bsk_rl/__init__.py` file, only remaining question is where `BSK_VERSION` should be specified. Needs to be in the src dir to check at import, which I think is where this check should happen given the development cycle.

```
from pkg_resources import (
    DistributionNotFound,
    VersionConflict,
    get_distribution,
    require,
)

BSK_VERSION = "2.2.1b0"
try:
    require(f"Basilisk>={BSK_VERSION}")
except VersionConflict:
    warn(
        f"Basilisk version {BSK_VERSION} or higher is required for full functionality. "
        f"Currently installed: {get_distribution('Basilisk').version}"
    )
except DistributionNotFound as e:
    e._template = (
        "The '{self.req}' distribution was not found "
        "and is required by {self.requirers_str}. Install from "
        "http://hanspeterschaub.info/basilisk/."
    )
    raise e
```
